extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/JpaOperations.java

```diff
@@ -71,16 +85,31 @@ public static boolean isPersistent(Object entity) {
     }
 
     public static void flush() {
-        delegate.flush();
+        Set<String> values = new HashSet<>(entityToPersistenceUnit.values());
+        for (String persistentUnit : values) {
+            getEntityManager(persistentUnit).flush();
```

Comment:
Shouldn't we only flush the entityManager of the entity?

Reply:
Yes, this is already done for https://github.com/quarkusio/quarkus/pull/8695/files#diff-2fcdac996e03f2fce93a3bc400488d93R70

The method is poorly named. I think it should be `flushAll`, there is a TODO comment in the code. 

https://github.com/quarkusio/quarkus/pull/8695/files#diff-2fcdac996e03f2fce93a3bc400488d93R103

Reply:
Or maybe I am missing something.

Reply:
Hmmm, I think the comment is on the wrong method then.

I think I would make `flush()` only flush the default persistence unit similar to what you did with other methods that are targeting the default if used without a parameter.

Reply:
So there are two methods:
- JpaOperations.flush(entity): this only flush the EntityManager of the entity
- JpaOperations.flush(): This flushes all the EntityManager. I was thinking that this one could be renamed to something like `flushAll()`, and certainly did not want to embark myself on that "refactoring things" journey.

Reply:
I would keep the current behavior and flush the default entity manager.

We can add a flushAll later if someone asks for it.

Reply:
Sure.

Reply:
@gsmet this is now done.